modules/order/src/Element/ProfileSelect.php

```diff
@@ -133,9 +264,33 @@ public static function validateForm(array &$element, FormStateInterface $form_st
    *   The current state of the form.
    */
   public static function submitForm(array &$element, FormStateInterface $form_state) {
-    $form_display = EntityFormDisplay::collectRenderDisplay($element['#profile'], 'default');
-    $form_display->extractFormValues($element['#profile'], $element, $form_state);
-    $element['#profile']->save();
+    $element_profile = $form_state->getValue($element['#parents']);
+
+    if ($element['#element_mode'] != 'view' && $form_state->isSubmitted()) {
+      $form_display = EntityFormDisplay::collectRenderDisplay($element_profile, 'default');
+      $form_display->extractFormValues($element_profile, $element, $form_state);
+      $element_profile->save();
+    }
+
+    $form_state->setValueForElement($element, $element_profile);
+  }
+
+  /**
+   * Ajax callback.
+   */
+  public static function ajaxRefresh(array &$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    $element = NestedArray::getValue($form, array_slice($triggering_element['#array_parents'], 0, -1));
+    return $element;
+  }
+
+  /**
+   * Ajax submit callback.
+   */
+  public static function ajaxSubmit(array &$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    $form_state->set('element_mode', $triggering_element['#element_mode']);
```

Comment:
As previous mention, the `element_mode` key in form state should be changed into:

```
    $parents = implode('-', $triggering_element['#parents']);
    $element_mode_name = 'element_mode-' . $parents;

    $form_state->set($element_mode_name, $triggering_element['#element_mode']);
```